x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.test.ts

```diff
@@ -0,0 +1,1006 @@
+        it('sets this to true', () => {
+          mount({
+            activeApiToken: {
+              ...newToken,
+              read: false,
+            },
+          });
+          CredentialsLogic.actions.setTokenType(PRIVATE);
+          expect(CredentialsLogic.values.activeApiToken.read).toEqual(true);
+        });
+      });
+      describe('when value is not PRIVATE', () => {
+        it('sets this to false', () => {
+          mount({
+            activeApiToken: {
+              ...newToken,
+              read: true,
+            },
+          });
+          CredentialsLogic.actions.setTokenType(ADMIN);
+          expect(CredentialsLogic.values.activeApiToken.read).toEqual(false);
+        });
+      });
+    });
+
+    describe('activeApiToken.type', () => {
+      it('sets the type value', () => {
+        mount({
+          activeApiToken: {
+            ...newToken,
+            type: ADMIN,
+          },
+        });
+        CredentialsLogic.actions.setTokenType(PRIVATE);
+        expect(CredentialsLogic.values.activeApiToken.type).toEqual(PRIVATE);
+      });
+    });
+
+    it('should not change any other values', () => {
+      mount();
+      CredentialsLogic.actions.setTokenType(PRIVATE);
+      expect(CredentialsLogic.values).toEqual({
+        ...DEFAULT_VALUES,
+        activeApiToken: expect.any(Object),
+      });
+    });
+  });
+
+  describe('toggleCredentialsForm', () => {
+    describe('showCredentialsForm', () => {
+      it('should toggle `showCredentialsForm`', () => {
+        mount({
+          showCredentialsForm: false,
+        });
+
+        CredentialsLogic.actions.toggleCredentialsForm();
+        expect(CredentialsLogic.values.showCredentialsForm).toEqual(true);
+
+        CredentialsLogic.actions.toggleCredentialsForm();
+        expect(CredentialsLogic.values.showCredentialsForm).toEqual(false);
+      });
+    });
+
+    describe('formErrors', () => {
+      it('should reset `formErrors`', () => {
+        mount({
+          formErrors: ['I am an error'],
+        });
+
+        CredentialsLogic.actions.toggleCredentialsForm();
+        expect(CredentialsLogic.values.formErrors).toEqual([]);
+      });
+    });
+
+    describe('activeApiTokenRawName', () => {
+      it('should set `activeApiTokenRawName` to the name of the provided token', () => {
+        mount();
+        CredentialsLogic.actions.toggleCredentialsForm(newToken);
+        expect(CredentialsLogic.values.activeApiTokenRawName).toEqual('myToken');
+      });
+
+      it('should set `activeApiTokenRawName` to the default value if no token is provided', () => {
+        mount();
+        CredentialsLogic.actions.toggleCredentialsForm();
+        expect(CredentialsLogic.values.activeApiTokenRawName).toEqual(
+          DEFAULT_VALUES.activeApiTokenRawName
+        );
+      });
+
+      // TODO: This fails, is this an issue? Instead of reseting back to the default value, it sets it to the previously
```

Comment:
Another oddity I found.

Reply:
Would be ++ for fixing. After some thought (maybe it's just very late) not totally sure how `activeApiTokenRawName` works as a selector tho - shouldn't `activeApiTokenName` be the selector instead? 🤷

Reply:
Since we know this works as is, and this PR is already big, I'd like to address this is a follow-up PR.

Reply:
Oh yeah, now that I think about it...

When a user enters a token name and then saves it, the token name is formatted before it's persisted to `activeApiToken.name`.

`activeApiTokenRawName` persists the unformatted name that the user had entered. I'm guessing this is used to back the form field.